frontend/src/components/FindMatch.js

```diff
@@ -134,6 +112,35 @@ const FindMatch = () => {
     )
   }
 
+  matchingSocket.on('found-connection', (username, difficulty) => {
+    const room = {
+      room_id: username,
+      id1: username,
+      id2: user.username,
+      id1_present: true,
+      id2_present: true,
+      difficulty: difficulty.toLowerCase(),
+      datetime: new Date(),
+    }
+    setRoom(room)
+  })
+
+  useEffect(async () => {
```

Comment:
`useEffect` doesn't support async functions directly and vscode should highlight this entire block yellow with a warning and show you the fix if you hover over it :D

For convenience, you'd usually use async functions in useEffect in this manner:
```
useEffect(() => {
  async function foo() {
    await something()
    ...
  }
  foo()
}, [...])

Reply:
ahhh i see, thank you. I thought that was an anonymouse function i could extend async to